### PR TITLE
Add support for reversedVPN proxy

### PR DESCRIPTION
**Pull request by mreiger**

### CODE REVIEW

File: pkg/webhook/controlplane/ensurer.go [hunk 1]
```diff
@@ -130,14 +153,23 @@ var (
...
 			ConfigMap: &corev1.ConfigMapVolumeSource{
 				LocalObjectReference: corev1.LocalObjectReference{Name: metal.AuditPolicyName},
 			},
 		},
 	}
+	customAuditPolicyVolume = corev1.Volume{
+		Name: metal.AuditPolicyName,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: "custom-audit-policy"},
```
Comment (majst01) [inline pkg/webhook/controlplane/ensurer.go:169]: maybe also a constant like metal.CustomAuditPolicyName

File: pkg/webhook/controlplane/ensurer.go [hunk 2]
```diff
@@ -337,45 +376,94 @@ func ensureKubeAPIServerCommandLineArgs(c *corev1.Container, makeAuditForwarder
...
+			logger.Info("AUDITDEBUG no custom auditpolicy configmap in shoot, deleting custom auditpolicy configmap in shoot namespace", "configmap", customAuditPolicyCm)
+			err := e.client.Delete(ctx, customAuditPolicyCm)
+			if err != nil {
+				return false, err
+			}
+		}
+		return false, nil
+	}
+
+	logger.Info("AUDITDEBUG found custom auditpolicy configmap in shoot", "shoot-configmap", customAuditPolicyShootCm)
```
Comment (majst01) [inline pkg/webhook/controlplane/ensurer.go:419]: log prefix is weird, simple remove the "AUDITDEBUG"

Comment (mreiger) [reply pkg/webhook/controlplane/ensurer.go]: The AUDITDEBUG messages were just for making sure the code does what I think it does - I'll pare them down to a less verbose logging when it's ready for review.

File: pkg/webhook/controlplane/ensurer.go (hunk 1 above)
Comment (mreiger) [reply pkg/webhook/controlplane/ensurer.go]: Will do that in the appropriate branch!

File: pkg/webhook/controlplane/ensurer.go [hunk 3]
```diff
@@ -338,44 +342,30 @@ func ensureKubeAPIServerCommandLineArgs(c *corev1.Container, makeAuditForwarder
 }
 
 func ensureAuditForwarder(ps *corev1.PodSpec, auditToSplunk bool) error {
+	auditForwarderSidecar := auditForwarderSidecarTemplate
```
Comment (Gerrit91) [inline pkg/webhook/controlplane/ensurer.go:345]: To be on the safe side, I'd call the DeepCopy function on the template:

```suggestion
	auditForwarderSidecar := auditForwarderSidecarTemplate.DeepCopy()
```

File: pkg/webhook/controlplane/ensurer.go [hunk 4]
```diff
@@ -200,38 +201,41 @@ var (
...
 			ReadOnly:  true,
 		},
 		{
 			Name:      "konnectivity-server-client-tls",
-			MountPath: "/konnectivity-proxy/client",
+			MountPath: "/proxy/client",
 			ReadOnly:  true,
 		},
 	}
-	konnectivityMtlsEnvVars = []corev1.EnvVar{
```
Comment (Gerrit91) [inline pkg/webhook/controlplane/ensurer.go:224]: Does this mean the support for konnectivity is dropped when rolling this out?

Comment (mreiger) [reply pkg/webhook/controlplane/ensurer.go:224]: No; konnectivity with mTLS is still supported; only the UDS proxy gets dropped.

The proxy environment variable is set in the ensureAuditForwarderProxy function, with the proxy host depending on which kind of proxy it is; therefore the global variable is not necessary any more.

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**